nvtabular/columns/schema.py

```diff
@@ -36,7 +137,32 @@ def with_tags(self, tags) -> "ColumnSchema":
 
         tags = list(set(list(self.tags) + tags))
 
-        return ColumnSchema(self.name, tags=tags)
+        return ColumnSchema(self.name, tags=tags, properties=self.properties)
+
+    def with_properties(self, properties):
+        if not isinstance(properties, dict):
+            raise TypeError("properties must be in dict format, key: value")
+
+        # Using new dictionary to avoid passing old ref to new schema
+        properties.update(self.properties)
+
+        return ColumnSchema(self.name, tags=self.tags, properties=properties)
+
+    def __eq__(self, other):
+        if not isinstance(other, ColumnSchema):
+            return False
+        if (
+            self.name == other.name
+            and self.tags == other.tags
+            and len(other.properties) == len(self.properties)
+            and self._is_list == other._is_list
+        ):
+            # iterate through to ensure ALL keys AND values equal
+            for prop_name, prop_value in other.properties.items():
+                if prop_name not in self.properties or self.properties[prop_name] != prop_value:
+                    return False
+            return True
```

Comment:
You can compare dictionaries with ==
```suggestion
            return self.properties == other.properties
```